aws/resource_aws_datasync_location_smb.go

```diff
@@ -0,0 +1,247 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/datasync"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+)
+
+func resourceAwsDataSyncLocationSmb() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsDataSyncLocationSmbCreate,
+		Read:   resourceAwsDataSyncLocationSmbRead,
+		Update: resourceAwsDataSyncLocationSmbUpdate,
+		Delete: resourceAwsDataSyncLocationSmbDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"agent_arns": {
+				Type:     schema.TypeSet,
+				Required: true,
+				ForceNew: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"domain": {
+				Type:     schema.TypeString,
+				Computed: true,
+				Optional: true,
+			},
+			"mount_options": {
+				Type:     schema.TypeList,
+				Optional: true,
+				ForceNew: true,
+				MaxItems: 1,
+				// Ignore missing config block (stolen from aws_ecs_service)
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
```

Comment:
Can use `suppressMissingOptionalConfigurationBlock` here instead of implementing inline.